openapi.yaml

```diff
@@ -1,190 +1,432 @@
-      attributes:
+        description: type of object, should be park
+        example: park
+      attributes: &parkAttributes
+        required: [name, location, amenities]
         properties:
           name:
             type: string
-            description: Name of pet
-            example: Hedwig
-          owner:
-            type: string
-            description: Name of owner
-            example: Harry Potter
-          species:
-            type: string
-            description: Species of pet
-            example: Owl
-  PetResult:
-    properties:
-      links:
-        $ref: '#/definitions/SelfLink'
-      data:
-        $ref: '#/definitions/PetResource'
-  PetsResult:
+            description: Name of Park
+            example: Avery Park
+          location:
+            type: object
+            required: [streetAddress, city, state, zip, latitude, longitude]
+            properties:
+              streetAddress:
+                type: string
+                description: Park address
+                example: '1200 SW Avery Park Dr.'
+              city:
+                type: string
+                description: city where the park is located
+                example: Corvallis
+              state:
+                type: string
+                description: Abbreviation of state where park is located
+                pattern: '^(?:(A[KLRZ]|C[AOT]|D[CE]|FL|GA|HI|I[ADLN]|K[SY]|LA|M[ADEINOST]|N[CDEHJMVY]|O[HKR]|P[AR]|RI|S[CD]|T[NX]|UT|V[AIT]|W[AIVY]))$'
+                example: OR
+              zip:
+                type: integer
+                description: Zip code where park is located
+                example: 97333
+              latitude:
+                type: number
+                format: float
+                description: latitude of park
+                example: 44.552283
+              longitude:
+                type: number
+                format: float
+                description: longitude of park
+                example: -123.269906
+          amenities:
+            type: array
+            items:
+              type: string
+            example: [play area, bike paths]
+            description: list of amenities at park
+      relationships:
+        type: object
+        properties:
+          data:
+            type: object
+            properties:
+              type:
+                type: string
+                example: owner
+              ownerId:
+                type: string
+                example: '1111'
+  PostParkModel:
+    required: [type, attributes]
     properties:
-      links:
-        allOf:
-          - $ref: '#/definitions/SelfLink'
-          - $ref: '#/definitions/PaginationLinks'
-      meta:
-        $ref: '#/definitions/Meta'
-      data:
-        type: array
-        items:
-          $ref: '#/definitions/PetResource'
+      type:
+        type: string
+        description: type of object, should be park
+        example: park
+      attributes: *parkAttributes
```

Comment:
I think `relationships` would also need to be included in the post body. Do you still have the business rule that a park must have an owner? If so, `relationships` would need to be required I think.

Also, if this is added, the `POST` endpoint will have to respond with a `404` if the owner isn't valid: https://jsonapi.org/format/1.1/#crud-creating-responses-404